tests/integration/e2e/test_basic_flow.py

```diff
@@ -209,8 +209,4 @@ async def test_consumed_messages(ops_test: OpsTest, deploy_data_integrator):
 
     check_produced_and_consumed_messages(uris, TOPIC)
 
-    await ops_test.model.applications[DATABASE_CHARM_NAME].remove()
```

Comment:
**suggestion, thoughts** I'm just wondering whether this should be conditioned on whether an external mongodb was provided or not. I mean: If we deploy a database (since the user did NOT pass the --database flag), we should probably tear the database down at the end.

Reply:
Updated the test to handle this case.